### ISSUE TITLE
Small tweak needed to build on macOS

### ISSUE THREAD

**Issue by dei-layborer** [JoseTomasTocino/freegemas#37]
Hi!

I was having an issue building on my Mac, but was able to fix it with a small tweak to ``CMakeLists.txt``.  Before, I was getting a linking error with regards to SDL2.  To fix it, I just added the following at line 178 (without replacing anything): ``find_package(SDL2 REQUIRED)``.

After this, it built fine (and runs).

My computer is an Apple Silicon Mac Mini (M1) with macOS 13.1.

**Comment by sharkwouter** [comment]
Thanks for the report! This is something I care about, but I have not been able to do any testing on MacOS, since I do not own a Mac. I'll try to see if I can get it to build on all supported platforms with this.

**Comment by dei-layborer** [comment]
Sure thing, glad I could help.  If you need any further testing on Mac, just let me know!

**Comment by sharkwouter** [comment]
@dei-layborer I've implemented the change and added automatic MacOS builds. Can you do a quick test to see if the resulting artifact works? You can find it here, it's called freegemas-macos: https://github.com/JoseTomasTocino/freegemas/actions/runs/4037791167

**Comment by sharkwouter** [comment]
That's a shame. I just installed sdl2 from brew like in the README, the code can be found here: https://github.com/JoseTomasTocino/freegemas/blob/8449000d5065673807990268cfa5e2479896b4e0/.github/workflows/build.yml#L60-L82

The chmod being needed I was expecting, Github does not save permissions for artifacts. Could you check what happens if you run `brew install cmake sdl2 sdl2_mixer sdl2_ttf sdl2_image gettext jsoncpp` and then try to run it?

Honestly, it would be best to package it better or to make a statically linked binary, but I personally have no way of testing that.

**Comment by dei-layborer** [comment]
I did that, but already had all of those installed (if from originally building it if nothing else).  Could it be related to the change I had to make to the CMakeLists to compile it myself?

I have far too little experience with C++ development to really know what the best approach would be for linking/packaging, I'm afraid.  But I'm happy to test things.

**Comment by Lewiscowles1986** [comment]
@dei-layborer check with latest instructions #39 solved this for me. Sorry for forgetting to tag in the PR @sharkwouter

**Comment by dei-layborer** [comment]
Hi!  Apologies for the delayed reply.  I just tried the new instructions (setting the backup library path), and everything builds properly!  I just get a couple warnings, but nothing fatal.  For your reference, I get the following when running ``cmake``:

```
CMake Warning (dev) at CMakeLists.txt:1 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

I do also get a couple C++ warnings when compiling (namely a variable being used uninitialized in an ``if`` condition).  There's also a linker warning saying that it's ignoring duplicate libraries:

``ld: warning: ignoring duplicate libraries: '-lSDL2', '-lSDL2_mixer', '-lSDL2_ttf', '-ljsoncpp'``

But again, everything works, and the game runs fine based on a quick test I just did.

I'm not closing the issue just in case, but everything seems fine on my end.
